routers/api/v1/repo/issue.go

```diff
@@ -410,14 +411,17 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
 	}
 
 	var deadlineUnix util.TimeStamp
+	var deadline time.Time
 	if form.Deadline != nil && !form.Deadline.IsZero() {
-		deadlineUnix = util.TimeStamp(form.Deadline.Unix())
+		deadline = time.Date(form.Deadline.Year(), form.Deadline.Month(), form.Deadline.Day(),
+			23, 59, 59, 0, form.Deadline.Location())
+		deadlineUnix = util.TimeStamp(deadline.Unix())
```

Comment:
I'd rather change the request Gitea makes to set the deadline instead of changing it here. Or we should'nt accept full datetime objects here anymore.

Reply:
I yet to see project where deadlines are planned by exact time. We should accept only date imho

Reply:
@lafriks If we do, we shouldn't accept full datetime objects, only dates.

Reply:
I agree that we shouldn't accept full datetime objects, however as we currently do I think that change should be in a new PR marked as breaking, perhaps against the 1.8 release.

@kolaente as this is a blocking PR for 1.7 release, would you be able to either accept changes as is, or give permission to dismiss your review?

Reply:
@techknowlogick ok, but can we at least add a notice in the swaggerdocs stating Gitea will ignore any passed time with the date and only use the date?

Reply:
@kolaente that sounds reasonable. I will create a PR for that in the SDK repo to add that to swaggerdocs.